### PR TITLE
Method AppRootCAsByChain() not thread safe

### PR DESCRIPTION
**Pull request by MayRosenbaum**
#### Type of change

- Bug fix

#### Description

As described in issue #4233 

#### Related issues

issue: #4233

### CODE REVIEW

File: internal/pkg/comm/connection.go [hunk 1]
```diff
@@ -75,10 +75,18 @@ func (cs *CredentialSupport) GetPeerCredentials() credentials.TransportCredentia
...
 	defer cs.mutex.RUnlock()
-	return cs.appRootCAsByChain
+	var trustedRoots [][]byte
+	// iterate over all roots for all app and orderer channels
+	for _, roots := range cs.appRootCAsByChain {
+		trustedRoots = append(trustedRoots, roots...)
+	}
+
+	trustedRootsCopy := make([][]byte, len(trustedRoots))
+	copy(trustedRootsCopy, trustedRoots)
```
Comment (C0rWin) [inline internal/pkg/comm/connection.go]:
```suggestion
	trustedRoots := make([][]byte, 0, len(cs.appRootCAsByChain))
	for _, roots := range cs.appRootCAsByChain {
		trustedRoots = append(trustedRoots, roots...)
	}

	return append([][]byte(nil), trustedRoots...)
}

Comment (C0rWin) [reply internal/pkg/comm/connection.go]: I think you can make it a little bit simpler.

Comment (C0rWin) [reply internal/pkg/comm/connection.go]: or even 

```
	trustedRoots := make([][]byte, 0, len(cs.appRootCAsByChain))
	for _, roots := range cs.appRootCAsByChain {
		trustedRoots = append(trustedRoots, roots...)
	}

	return trustedRoots
```

Comment (MayRosenbaum) [reply internal/pkg/comm/connection.go]: Fixed to the second option

**Review by C0rWin — APPROVED** [review]

**Pull request merged**